datahub-web-react/src/app/applications/ApplicationsTable.tsx

```diff
@@ -0,0 +1,220 @@
+                const nameB = entityRegistry.getDisplayName(EntityType.Application, b.entity);
+                return nameA.localeCompare(nameB);
+            });
+    }, [applicationsData, searchQuery, entityRegistry]);
+
+    const isLoading = propLoading || networkStatus === NetworkStatus.refetch;
+
+    // Simplified function to initiate tag deletion
+    const showDeleteConfirmation = useCallback(
+        (applicationUrn: string) => {
+            // Find the application entity from applicationsData
+            const applicationData = applicationsData.find((result) => result.entity.urn === applicationUrn);
+            if (!applicationData) {
+                message.error('Failed to find application information');
+                return;
+            }
+
+            const fullDisplayName = entityRegistry.getDisplayName(EntityType.Application, applicationData.entity);
+
+            setApplicationUrnToDelete(applicationUrn);
+            setApplicationDisplayName(fullDisplayName);
+            setShowDeleteModal(true);
+        },
+        [entityRegistry, applicationsData],
+    );
+
+    // Function to handle the actual application deletion
+    const handleDeleteApplication = useCallback(() => {
+        deleteApplicationMutation({
+            variables: {
+                urn: applicationUrnToDelete,
+            },
+        })
+            .then(() => {
+                message.success(`Application "${applicationDisplayName}" has been deleted`);
+                refetch(); // Refresh the application list
+            })
+            .catch((e: any) => {
+                message.error(`Failed to delete tag: ${e.message}`);
+            });
+
+        setShowDeleteModal(false);
+    }, [deleteApplicationMutation, refetch, applicationUrnToDelete, applicationDisplayName]);
+
+    const columns = useMemo(
+        () => [
+            {
+                title: 'Application',
+                key: 'application',
+                render: (record) => {
+                    const application = record.entity;
+                    const displayName = entityRegistry.getDisplayName(EntityType.Application, application);
+                    return (
+                        <ApplicationNameColumn
+                            applicationUrn={application.urn}
+                            displayName={displayName}
+                            searchQuery={searchQuery}
+                        />
+                    );
+                },
+                sorter: (sourceA, sourceB) => {
+                    const nameA = entityRegistry.getDisplayName(EntityType.Application, sourceA.entity);
+                    const nameB = entityRegistry.getDisplayName(EntityType.Application, sourceB.entity);
+                    return nameA.localeCompare(nameB);
+                },
+                sortOrder: sortedInfo.columnKey === 'tag' ? sortedInfo.order : null,
```

Comment:
the issue with this sorting is that it's done on the frontend and if we are paginating on the backend then we would only sort on the current page you're on instead of across all applications as the user would expect